src/components/SingleArticle.jsx

```diff
@@ -0,0 +1,38 @@
+import { useEffect, useState } from "react";
+import { getArticleById } from "../api";
+import { useParams } from "react-router-dom";
+import SingleArticleCard from "./SingleArticleCard";
+
+function SingleArticle() {
+  const [article, setArticle] = useState({});
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    getArticleById(article_id)
+      .then((article) => {
+        setArticle(article);
+        setIsLoading(false);
+      })
+      .catch((err) => {
+        console.log(err);
+        setError(true);
+        setIsLoading(false);
+      });
+  }, [article_id]);
+
+  return (
+    <>
+      {isLoading ? (
+        <p>Articles are loading...</p>
+      ) : error ? (
+        <p>Oops, something went wrong! Please try again...</p>
```

Comment:
Likewise with your error messages - great opportunity to demonstrate a reusable component.